### PR TITLE
SUS-810: Filearchive - get username from id in ArchivedFile, SpecialUndelete and ApiQueryFilearchive

### PR DESCRIPTION
**Pull request by saipetch**
SUS-810:
-  get username from id for filearchive in ArchivedFile, SpecialUndelete and ApiQueryFilearchive

UPDATE (pwojcik): I updated the PR with the lookup feature added in SUS-1011

### CODE REVIEW

**Comment by ghost** [comment]
I have added logging with sapling of 1%. I will check the existing logs to estimate the usage.

File: includes/api/ApiQueryFilearchive.php [hunk 1]
```diff
@@ -145,7 +145,19 @@ public function execute() {
 			}
 			if ( $fld_user ) {
 				$file['userid'] = $row->fa_user;
-				$file['user'] = $row->fa_user_text;
+				/* Wikia change begin */
```
Comment (ghost) [inline includes/api/ApiQueryFilearchive.php:148]: Currently, the code here is executed 200 times per hour.

File: includes/filerepo/file/ArchivedFile.php [hunk 2]
```diff
@@ -163,7 +163,15 @@ public function load() {
 			$this->media_type = $row->fa_media_type;
 			$this->description = $row->fa_description;
 			$this->user = $row->fa_user;
-			$this->user_text = $row->fa_user_text;
+			/* Wikia change begin */
```
Comment (ghost) [inline includes/filerepo/file/ArchivedFile.php:167]: Currently, the code here is executed up to 60 times per hour.

File: includes/specials/SpecialUndelete.php [hunk 3]
```diff
@@ -1134,10 +1134,19 @@ private function showHistory() {
 		}
 		if( $haveFiles ) {
 			$batch = new LinkBatch();
+			/* Wikia change begin */
 			foreach ( $files as $row ) {
-				$batch->addObj( Title::makeTitleSafe( NS_USER, $row->fa_user_text ) );
-				$batch->addObj( Title::makeTitleSafe( NS_USER_TALK, $row->fa_user_text ) );
+				if ( ( new Wikia\Util\Statistics\BernoulliTrial( 0.01 ) )->shouldSample() ) {
```
Comment (ghost) [inline includes/specials/SpecialUndelete.php]: `Special:Undelete` can trigger pretty heavy traffic spikes. We should definitely keep an eye on it and release the username-related changes separately.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 120 issues
* MAJOR 17 major
* MINOR 64 minor
* INFO 39 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ApiQueryFilearchive.php#L40](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/api/ApiQueryFilearchive.php#L40): The Cyclomatic Complexity of this function "execute" is 36 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [ApiQueryFilearchive.php#L40](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/api/ApiQueryFilearchive.php#L40): This function "execute" has 166 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
1. MAJOR [ApiQueryFilearchive.php#L128](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/api/ApiQueryFilearchive.php#L128): Extract this increment or decrement operator into a dedicated statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS881)
1. MAJOR [ApiQueryFilearchive.php#L157](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/api/ApiQueryFilearchive.php#L157): Remove this unused "$vals" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [ArchivedFile.php#L14](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/filerepo/file/ArchivedFile.php#L14): Class "ArchivedFile" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [SpecialUndelete.php](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/specials/SpecialUndelete.php): File "SpecialUndelete.php" has 1499 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
1. MAJOR [SpecialUndelete.php#L328](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/specials/SpecialUndelete.php#L328): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/specials/SpecialUndelete.php#L411): The Cyclomatic Complexity of this function "undeleteRevisions" is 31 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/specials/SpecialUndelete.php#L411): Reduce the number of returns of this function 8, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/87f631128c1d736aaec277858dd1cbe8685d4882/includes/specials/SpecialUndelete.php#L411): This function "undeleteRevisions" has 229 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)

**Comment by jcellary** [comment]
LGTM apart from https://github.com/Wikia/app/pull/11015/files#diff-9d3e282519bf53cecc0cd737b595d9a7R116 - I don't think we need any extra logging as we added it to the getUsername function

**Comment by WikiaTech** [comment]
SonarQube analysis reported 120 issues
- MAJOR 17 major
- MINOR 64 minor
- INFO 39 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [ApiQueryFilearchive.php#L40](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/api/ApiQueryFilearchive.php#L40): The Cyclomatic Complexity of this function "execute" is 36 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
2. MAJOR [ApiQueryFilearchive.php#L40](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/api/ApiQueryFilearchive.php#L40): This function "execute" has 166 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
3. MAJOR [ApiQueryFilearchive.php#L128](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/api/ApiQueryFilearchive.php#L128): Extract this increment or decrement operator into a dedicated statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS881)
4. MAJOR [ApiQueryFilearchive.php#L157](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/api/ApiQueryFilearchive.php#L157): Remove this unused "$vals" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
5. MAJOR [ArchivedFile.php#L14](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/filerepo/file/ArchivedFile.php#L14): Class "ArchivedFile" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
6. MAJOR [SpecialUndelete.php](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/specials/SpecialUndelete.php): File "SpecialUndelete.php" has 1499 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
7. MAJOR [SpecialUndelete.php#L328](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/specials/SpecialUndelete.php#L328): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
8. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/specials/SpecialUndelete.php#L411): The Cyclomatic Complexity of this function "undeleteRevisions" is 31 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
9. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/specials/SpecialUndelete.php#L411): Reduce the number of returns of this function 8, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
10. MAJOR [SpecialUndelete.php#L411](https://github.com/Wikia/app/blob/410634275823e284ed2b7ad7a87e80c15c242feb/includes/specials/SpecialUndelete.php#L411): This function "undeleteRevisions" has 229 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)

**Pull request merged**